lib/tlscontext.c

```diff
@@ -313,6 +314,21 @@ tls_session_info_callback(const SSL *ssl, int where, int ret)
     }
 }
 
+static gboolean
+_set_sni_in_client_mode(TLSSession *self)
+{
+  if (!self->ctx->sni)
+    return TRUE;
+
+  if (self->ctx->mode != TM_CLIENT)
+    return TRUE;
+
+  if (SSL_set_tlsext_host_name(self->ssl, self->ctx->sni))
```

Comment:
Note that `SSL_set_tlsext_host_name()` is documented only for OpenSSL 1.1.1:
https://www.openssl.org/docs/man1.1.1/man3/SSL_get_servername_type.html

It exists in 1.0.2 and 1.1.0 without docs, so I guess it's OK, no compat layer is needed.

Reply:
Yes, that is what I have seen. It is a fairly old code, not sure why the documentation was behind so much,